galsim/hsm.py

```diff
@@ -220,6 +220,82 @@ def __getstate__(self):
     def __setstate__(self, state):
         self.__init__(*state)
 
+    def applyWCS(self, wcs, image_pos=None):
+        """Convert moments in pixel coordinates to moments in sky coordinates.
+
+        Natively, the HSM algorithm computes second moments in (x,y) coordinates in the
+        sensor coordinate system.  However, many applications of second moments for shape
+        measurements need the measurements in sky coordinates.  This method converts the
+        moments to (u,v) coordinates, where +v is towards North and +u is towards West.
+
+        The values that are different from the original are:
+
+        moments_sigma is changed to be in units of arcsec.
+        observed_shape is changed to be in (u,v) coordinates.
+        moments_centroid is changed to be in (u,v) coordinates in units of arcsec.
+        moments_rho4 is changed to be in units of arcsec^4.
+
+        .. note::
```

Comment:
I think I'd rather invalidate (set to NaN or None?) the untouched parameters.  Having an object where `.moments_sigma` is in arcsec but `.psf_sigma` is in pixels or `.moments_shape` and `.psf_shape` are in different coordinate systems feels error prone.

Reply:
Sure.  Sounds reasonable to me. I changed it to reset any EstimateShear values to defaults.